test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -747,6 +755,83 @@ def __are_requirements_matching(param_dic=None, test=None):
             if not eff_tc_req['no_comm_on_mom']:
                 return False
 
+    def hardware_report(self):
+        """
+        function used to get hardware report
+        every 5 minutes
+        """
+        du = DshUtils()
+        self.hardware_report_timer = Timer(300, self.hardware_report)
+        self.hardware_report_timer.start()
+        self.hardware_monitor_start = True
+        for system in ['servers', 'moms', 'comms']:
```

Comment:
There will be chances of a duplicate host in servers, moms and comms so you can optimize these two for loop with the below code.
```
systems = self.param_dict['servers'].host_keys()
systems.extend(self.param_dict['moms'].host_keys())
systems.extend(self.param_dict['comms'].host_keys())
systems = list(set(systems))
for system in systems:
   ...
```

Reply:
moms doesn't have server_priv, comm_priv directory if all are a different system. So i have to check according to moms, servers and comms. But all or two on one system i have to use one other condition.

Reply:
as per offline discussion, you can make use of du.isdir() to check whether require directory exists or not then check for the files in that directory